### DISCUSSION TITLE
Why `-` is not allowed in app name

### DISCUSSION THREAD

**Discussion by k4wo** [nrwl/nx#28494] [General]
What is a reason `-` is not allowed in a name of remote app in module federation? For host and library is not prohibited. I've tested it by eg.:

```
npx nx g @nx/react:host apps/portal-ui --remotes=b-ui,i-ui,p_ui --bundler=rspack --dynamic
```

I've noticed it in 20.0.1 version, it's weird because in the past `remotes` were allowed to use `-` in name.

**Comment by k4wo** [comment]
@Coly010 are you able to give us more details?

**Comment by Coly010** [comment]
Remotes loaded using `remoteType` module are technically able to use `-`.

However, Node Federation doesn’t like `remoteType=module` and React doesn’t really make sense to be forcing into ESM with Webpack and Rspack. So it uses the default remote loading option. 

This requires setting a variable at runtime. That variable is given the same name as the remote. (Module federation controls this, not Nx).

Therefore, we’re now enforcing that remote names align with JS variable naming restrictions, as this is what happens at runtime.
